### PR TITLE
Request Ride Page

### PR DESCRIPTION
**Pull request by tyang98**
### Summary 

Created the form for requesting rides as well as the pages for both pickup and drop-off.

### Notes 

Need to find a way to transition from page to page

### Screenshots

Screen Shot 2019-12-14 at 10 17 41 PM
Screen Shot 2019-12-14 at 10 17 11 PM
Screen Shot 2019-12-14 at 10 16 43 PM

### CODE REVIEW

File: Rider/app/build.gradle [hunk 1]
```diff
@@ -20,15 +20,27 @@ android {
...
-    implementation 'androidx.core:core-ktx:1.0.2'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.core:core-ktx:1.1.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.android.support:design:28.0.0'
+    implementation 'com.android.support:recyclerview-v7:28.0.0'
+    implementation 'com.android.support:cardview-v7:28.0.0'
+    implementation 'com.google.android.gms:play-services-auth:17.0.0'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (davidchuyaya) [inline Rider/app/build.gradle:36]: why?

File: Rider/app/build.gradle [hunk 2]
```diff
@@ -20,15 +20,27 @@ android {
...
+    implementation 'de.hdodenhof:circleimageview:3.0.1'
+    implementation 'androidx.preference:preference:1.1.0-alpha05'
+    implementation 'com.github.bumptech.glide:glide:4.10.0'
+    implementation group: 'org.apache.httpcomponents', name: 'httpclient-android', version: '4.3.5.1'
+    implementation "cz.msebera.android:httpclient:4.4.1.2"
```
Comment (davidchuyaya) [inline Rider/app/build.gradle]: probably don't need this; research how to do internet access in Android with Volley (or look at cue/o-week)

File: Rider/app/build.gradle [hunk 3]
```diff
@@ -20,15 +20,27 @@ android {
...
+    annotationProcessor 'com.github.bumptech.glide:compiler:4.10.0'
     testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.0'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
```
Comment (davidchuyaya) [inline Rider/app/build.gradle]: remove if there are no tests

File: Rider/app/build.gradle [hunk 4]
```diff
@@ -20,15 +20,27 @@ android {
...
-    implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'androidx.appcompat:appcompat:1.0.2'
-    implementation 'androidx.core:core-ktx:1.0.2'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.core:core-ktx:1.1.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.android.support:design:28.0.0'
+    implementation 'com.android.support:recyclerview-v7:28.0.0'
+    implementation 'com.android.support:cardview-v7:28.0.0'
```
Comment (davidchuyaya) [inline Rider/app/build.gradle]: Use androidx and the material design components. Lmk if you don't know how to migrate to those.

File: Rider/app/build.gradle [hunk 5]
```diff
@@ -20,15 +20,27 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    useLibrary 'org.apache.http.legacy'
```
Comment (davidchuyaya) [inline Rider/app/build.gradle]: Probably a bad idea

File: Rider/app/build.gradle [hunk 6]
```diff
@@ -7,7 +7,7 @@ apply plugin: 'kotlin-android-extensions'
 android {
     compileSdkVersion 28
     defaultConfig {
-        applicationId "com.example.rider"
+        applicationId "com.dti.rider"
         minSdkVersion 15
```
Comment (davidchuyaya) [inline Rider/app/build.gradle]: This is way too low and means lots of functionalities will be unavailable; at least have Lollipop

File: Rider/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -9,13 +11,35 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name=".DropOffActivity"></activity>
+        <activity android:name=".PickupActivity" />
```
Comment (davidchuyaya) [inline Rider/app/src/main/AndroidManifest.xml]: Consistency

File: Rider/app/src/main/java/com/dti/rider/LoginActivity.kt [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+                val statusCode = response.statusLine.statusCode
+                val responseBody = EntityUtils.toString(response.entity)
+                Log.i("Successful Sign In", "Signed in as: $responseBody")
+                startActivity(Intent(this, MainActivity::class.java))
+
+            } catch (e: ClientProtocolException) {
+                Log.e("Token Error", "Error sending ID token to backend.", e)
+            } catch (e: IOException) {
+                Log.e("Token Error", "Error sending ID token to backend.", e)
+            }
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/LoginActivity.kt]: If you don't distinguish the errors, just catch a general exception

File: Rider/app/src/main/java/com/dti/rider/LoginActivity.kt [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+        } catch (e: ApiException) {
+            Log.w("Google Sign In Error", "signInResult:failed code: " + e.statusCode)
+            Log.w(
+                "Google Sign In Error",
+                "signInResult:failed message: " + GoogleSignInStatusCodes.getStatusCodeString(e.statusCode)
+            )
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/LoginActivity.kt]: Not sure why these are separate

File: Rider/app/src/main/java/com/dti/rider/LoginActivity.kt [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+            Toast.makeText(this, "Failed", Toast.LENGTH_LONG).show()
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/LoginActivity.kt]: Not the best UX for error messages

Comment (davidchuyaya) [reply Rider/app/src/main/java/com/dti/rider/LoginActivity.kt]: Also don't forget about string resources

File: Rider/app/src/main/java/com/dti/rider/MainActivity.kt [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        btn_pickup.setOnClickListener {
+            val intent = Intent(this, PickupActivity::class.java)
+            startActivity(intent)
+        }
+
+        val btn_dropoff = findViewById(R.id.dropOffButton) as Button
+        btn_dropoff.setOnClickListener {
+            val intent = Intent(this, DropOffActivity::class.java)
+            startActivity(intent)
+        }
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/MainActivity.kt]: DRY

File: Rider/app/src/main/java/com/dti/rider/RequestActivity.kt [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+        wcCheckBox = findViewById(R.id.wheelchair)
+
+    }
+
+    private fun Check(v: View) {
+        if (wcCheckBox.isChecked()) {
+            val wcMsg = "Wheelchair"
+            Toast.makeText(
+                this, wcMsg + "is selected",
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/RequestActivity.kt]: String resources

File: Rider/app/src/main/java/com/dti/rider/RequestActivity.kt [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+class RequestActivity : AppCompatActivity() {
+
+
+    private lateinit var wcCheckBox: CheckBox
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_request)
+
+        getSupportActionBar()?.setTitle(Html.fromHtml("<font color='#FFFFFF'>Schedule</font>"))
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/RequestActivity.kt]: What the heck is happening here?

File: Rider/app/src/main/java/com/dti/rider/ProfileActivity.kt [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+    private lateinit var mGoogleSignInClient: GoogleSignInClient
+    private lateinit var name: TextView
+    private lateinit var email: TextView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_profile)
+
+        getSupportActionBar()?.setTitle(Html.fromHtml("<font color='#FFFFFF'>Schedule</font>"))
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/ProfileActivity.kt]: Huh??

File: Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+    private lateinit var mGoogleSignInClient: GoogleSignInClient
+    private lateinit var name: TextView
+    private lateinit var email: TextView
+    private lateinit var signOutButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings)
+
+        getSupportActionBar()?.setTitle(Html.fromHtml("<font color='#070707'>Schedule</font>"))
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt]: Please no more (and DRY)

File: Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+        val acct = GoogleSignIn.getLastSignedInAccount(this)
+
+        if (acct != null) {
+            val personName = acct.displayName
+            val personEmail = acct.email
+
+            name.setText(personName)
+            email.setText(personEmail)
+        }
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt:45]: sure looks familiar

File: Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+            signOutButton.visibility = View.VISIBLE
+        } else {
+            signOutButton.visibility = View.GONE
+        }
+    }
+
+    private fun signOut() {
+        mGoogleSignInClient.signOut()
+            .addOnCompleteListener(this) {
+                Toast.makeText(this, "Signed Out Successfully", Toast.LENGTH_SHORT).show()
```
Comment (davidchuyaya) [inline Rider/app/src/main/java/com/dti/rider/SettingsActivity.kt]: String resource

File: Rider/app/src/main/res/drawable/edittext_back.xml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="@color/eTextColor" />
+
+    <corners
+        android:radius="5dp" />
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/drawable/edittext_back.xml:7]: dimen resource

File: Rider/app/src/main/res/drawable/et_style.xml [hunk 19]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+
+    <stroke
+        android:width="2dp"
+        android:color="#DBDBDB" />
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/drawable/et_style.xml:7]: resource

File: Rider/app/src/main/res/layout/activity_dropoff.xml [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".DropOffActivity">
+
+    <TextView
+        android:id="@+id/cancelText"
+        style="@style/TextAppearance.AppCompat.Medium"
+        android:layout_width="55dp"
+        android:layout_height="20dp"
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_dropoff.xml:13]: Text uses sp, not dp. Review my Android slides if that helps

File: Rider/app/src/main/res/layout/activity_dropoff.xml [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+        android:text="@string/cancel"
+        android:textColor="@color/blackColor"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toTopOf="parent"/>
+
+    <TextView
+        android:id="@+id/dropOffText"
+        style="@style/TextAppearance.AppCompat.Large"
+        android:layout_width="wrap_content"
+        android:layout_height="26dp"
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_dropoff.xml:25]: Don't set height for text

File: Rider/app/src/main/res/layout/activity_dropoff.xml [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginLeft="90dp"
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_dropoff.xml]: You seem to switch between resources & raw numbers a lot

File: Rider/app/src/main/res/layout/activity_main.xml [hunk 23]
```diff
@@ -1,18 +1,79 @@
...
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".MainActivity">
 
-    <TextView
-        android:layout_width="wrap_content"
+    <LinearLayout
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_main.xml]: Keep the ConstraintLayout if possible; it's the newly encouraged paradigm

File: Rider/app/src/main/res/layout/activity_profile.xml [hunk 24]
```diff
@@ -0,0 +1,219 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="@dimen/text_margin"
+                android:layout_marginLeft="@dimen/image_margin"
+                android:layout_marginTop="@dimen/split_margin"
+                android:drawableLeft="@drawable/ic_person_black_24dp"
+                android:drawableRight="@drawable/ic_chevron_right_black_24dp"
+                android:drawablePadding="@dimen/image_margin"
+                android:gravity="center_vertical"
+                android:text="How should we address you?"
+                app:layout_constraintTop_toTopOf="parent"></TextView>
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_profile.xml]: Why </TextView>?

File: Rider/app/src/main/res/layout/activity_settings.xml [hunk 25]
```diff
@@ -0,0 +1,244 @@
...
+            android:background="@color/profileColor"
+            android:contentDescription="@string/settings"
+            android:text="@string/settings"
+            android:textColor="@color/blackColor"
+            android:textStyle="bold" />
+
+    </androidx.cardview.widget.CardView>
+
+
+    <androidx.cardview.widget.CardView
```
Comment (davidchuyaya) [inline Rider/app/src/main/res/layout/activity_settings.xml]: No hardcoding please

File: Rider/gradle/wrapper/gradle-wrapper.properties [hunk 26]
```diff
@@ -1,4 +1,4 @@
-#Tue Oct 22 17:31:47 EDT 2019
+#Sun Oct 27 00:36:55 EDT 2019
```
Comment (davidchuyaya) [inline Rider/gradle/wrapper/gradle-wrapper.properties:1]: Perhaps this file should also not be committed

**Review by davidchuyaya — CHANGES_REQUESTED** [review]

File: Rider/gradle/wrapper/gradle-wrapper.properties (hunk 26 above)
Comment (SamChou19815) [reply Rider/gradle/wrapper/gradle-wrapper.properties:1]: Gradle wrapper is designed to be committed. We should probably avoid unnecessary changes here.

**Pull request closed without merging**